src/makeCalculator.js

```diff
@@ -37,7 +37,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(action, num) {
+      action(num, this);
+
+      return this;
+    },
+
+    add(num, currentObj) {
+      currentObj.result += num;
```

Comment:
and here u just can get result

Reply:
subtract(num, result) {
      result -= num;
    },
    
but i reassign param

Reply:
> function should return calculation

subtract(num, result) {
result -= num;
},

but i reassign param

Reply:
you return from operator, so it's ok